### ISSUE TITLE
status.proto in python (can't get full details)

### ISSUE THREAD

**Issue by moelius** [grpc/grpc#16366]
### What version of gRPC and what language are you using?
 1.14.1 python
 
### What operating system (Linux, Windows, …) and version?
 MacOS HighSierra 10.13.6
 
### What runtime / compiler are you using (e.g. python version or version of gcc)
 python 3.6.5
 
### What did you do?
If possible, provide a recipe for reproducing the error. Try being specific and include code snippets if helpful.
 In golang code I use custom status.ptoto details:

**proto**:
```
message EnumRequestType {
    enum Request {
        ResolveUsername = 0;
        GetFullChannel = 1;
        GetHistory = 2;
        GetMessages = 3;
        GetMessagesViews = 4;
        Unknown = 5;
    }
}

message EnumNotFoundDetailCode {
    enum Code {
        SUBSCRIPTION_NOT_EXISTS = 0;
        SESSION_NOT_EXISTS = 1;
        LOCKED = 2;
        BANNED = 3;
        RATE_LIMIT_EXCEEDED = 4;
    }
}

message RateLimitExceededDetails {
    uint64 phone = 1;
    EnumRequestType.Request request = 2;
    string rate_limit = 3;
    int64 reset_time_left = 4; //milliseconds
}

message NotFoundDetails {
    EnumNotFoundDetailCode.Code code = 1;
    oneof details {
        RateLimitExceededDetails rate_limit_exceeded_details = 2;
    }
}
```

**go server code**:

```golang
if err == al.ErrSessionLocked {
	if st, e := status.New(codes.NotFound, err.Error()).WithDetails(&grpcgrievoustlg.NotFoundDetails{Code: grpcgrievoustlg.EnumNotFoundDetailCode_LOCKED}); e != nil {
		err = e
	} else {
		err = st.Err()
	}
}
```
**go client code**:

```golang
if st, ok := status.FromError(err); !ok {
	log.Println(st.Err())
} else if st.Code() != codes.OK {
	if st.Code() == codes.NotFound {
		details := st.Details()[0].(*grpcgrievoustlg.NotFoundDetails)
		switch details.Code {
		case grpcgrievoustlg.EnumNotFoundDetailCode_RATE_LIMIT_EXCEEDED:
			log.Println(st.Err())
		default:
			log.Println(st.Err())
		}
	} else {
		log.Println(st.Err())
	}
}
```
In python I only have a representation string of this details.

**python client code**:

```python
if e.code() != grpc.StatusCode.NOT_FOUND:
    raise SessionManagerException(e)
details=e.details()
```
### What did you expect to see?
 
 Full protobuf `NotFoundDetails` message
### What did you see instead?
 `'rate limit exceeded  [phone: 79166667818, request: GetHistory, rate limit: 1_per_second, reset time left: 973ms]'`
Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).
 
### Anything else we should know about your project / environment?

No

**Comment by tombry** [comment]
I'm just getting started with gRPC, Python, and proto3, and I also encountered this problem.  We were planning to leverage https://godoc.org/google.golang.org/grpc/status#Status.Details, and it looks like adding details to an error response is also a problem for gRPC _services_ written in Python.  

I see that this ticket was accepted as a bug last week.  I don't know how big this bug is, but it would be helpful to understand the timeframe when we might see a fix.  That will help my team decide whether we should assume that we'll have support for Details as we start to build out our gRPC services.

Thanks.

**Comment by moelius** [comment]
HI, temporarily, I solve this problem, preparing a error on golang side, for later parsing on the side of python. But I don't like this solution =(

**Comment by etiennebch** [comment]
Hello, I think I have encountered the issue if I understand your problem correctly. The service was written in Go and the client in Python. [Here is a post I did on Medium](https://medium.com/kuranda-labs-engineering/gracefully-handling-grpc-errors-in-a-go-server-python-client-setup-9805a5464692) to explain how to work around the issue. You should be able to leverage the function `trailing_metadata` of your gRPC response object in Python.

**Comment by lidizheng** [comment]
Blocked by enigmatic issue with `googleapis-common-protos` and `Bazel`...
Import order will affect build result for Python https://github.com/bazelbuild/bazel/issues/6844

**Comment by lidizheng** [comment]
This feature has been published at `v1.18.0`.
For DEMO snippet of how to use it, please check https://github.com/grpc/grpc/issues/17601#issuecomment-450312923.
